### ISSUE TITLE
date() ignores offset?

### ISSUE THREAD

**Issue by neumino** [rethinkdb/docs#744]
These two commands return the same dates (same epoch_time field but different timezone field).

```
r.time(1986, 11, 3, '+02:00').date()
r.time(1986, 11, 3, '+00:00').date()
```

I thought that date would return the date given the timezone, but it looks like it just return the date for the offset `+00:00`. 
If it's expected, a little note on the docs would be nice : )

**Comment by neumino** [comment]
Same question for timeOfDay.

**Comment by danielmewes** [comment]
Hmm it's working for me on 2.0.0:
```
r.time(1986, 11, 3, "+00:00").toEpochTime()
531360000
r.time(1986, 11, 3, "+02:00").toEpochTime()
531352800
```

Which outputs are you getting?

**Comment by danielmewes** [comment]
Oh never mind. The issue is with `date`. I see.

**Comment by danielmewes** [comment]
@mlucy was this the intended behavior of `date`?

**Comment by mlucy** [comment]
I don't remember what the intended behavior was, but if this is confusing I'm not at all opposed to changing it.

**Comment by danielmewes** [comment]
I opened https://github.com/rethinkdb/rethinkdb/issues/4149 . I think this might have been an oversight in the implementation, since with the current implementation this has some very fishy side-effects.

**Comment by mlucy** [comment]
@danielmewes -- should we close this one then?

**Comment by danielmewes** [comment]
Yeah. We'll fix it with 2.1 in the server.
